### PR TITLE
Sdk perf compare

### PR DESCRIPTION
**Pull request by sophiaso**

### CODE REVIEW

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    private static final int TIMES = 100;
+    private static final int BATCH_SIZE = 10;
+    private static final float ACCEPTANCE_DIFF = 0.1f;
+    private static boolean hasInit = false;
+    private static SdkService sdkService;
+    private static PerformanceReport report = new PerformanceReport();
+
+    @Before
+    public void setup() throws DocumentClientException {
+        if (!hasInit) {
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: seems you may need `@BeforeClass` ? Please correct me if wrong.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+    @Test
+    public void saveOneRecordTest() {
+        final List<PerfPerson> personList = PerfDataProvider.getPerfData(TIMES);
+
+        final long springCost = applyInputListFunc(personList, repository::save);
+        final long sdkCost = applyInputListFunc(personList, sdkService::save);
+
+        report.addItem(new PerfItem(OperationType.SAVE_ONE, springCost, sdkCost, TIMES));
+        assertPerf(springCost, sdkCost, ACCEPTANCE_DIFF);
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: line 83 and 84 seems code pattern except OperationType, looks we can add one method like completeTest to avoiding duplicated code.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+    @Test
+    public void countTest() {
+        final long springCost = getSupplier(TIMES, repository::count);
+        final long sdkCost = getSupplier(TIMES, sdkService::count);
+
+        report.addItem(new PerfItem(OperationType.COUNT, springCost, sdkCost, TIMES));
+        assertPerf(springCost, sdkCost, ACCEPTANCE_DIFF);
+    }
+
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: duplicated empty line.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        applyInputListFunc(personList, repository::saveAll);
+        applyInputListFunc(personList, sdkService::saveAll);
+
+        return personList;
+    }
+
+    private List<Iterable<String>> listBatchIds(int times, int batchSize) {
+        return prepareListBatchData(times, batchSize).stream()
+                .map(iterable -> {
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: looks can be done like `.stream().map(Lists::newArrayList).flatmap(List::stream).map(Person::getId()).Collect(Collectors.toList())`

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceReport.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+package com.microsoft.azure.spring.data.cosmosdb.performance;
+
+import com.google.common.collect.Lists;
+import lombok.Data;
+
+import java.util.List;
+
+public class PerformanceReport {
+    private static final String NEW_LINE = System.lineSeparator();
+    private List<PerfItem> perfItems = Lists.newArrayList();
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceReport.java]: final ?

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceReport.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    public void addItem(PerfItem item) {
+        perfItems.add(item);
+    }
+
+    public List<PerfItem> getPerfItems() {
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceReport.java]: `@Getter` ?

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java (hunk 1 above)
Comment (sophiaso) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: `@BeforeClass` requires static  method, which cannot use the Autowired non-static variable.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java (hunk 2 above)
Comment (sophiaso) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: Updated.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java (hunk 3 above)
Comment (sophiaso) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java]: Removed.

**Pull request merged**